src/relay/backend/contrib/ethosn/codegen_ethosn.h

```diff
@@ -0,0 +1,331 @@
+    idx = 0;
+    for (const auto& layer : operand_table_[func->body]) {
+      AddOutput(network_, *layer);
+      network_with_ids.output_ids[id_table_[func->body][idx]] = idx;
+      idx++;
+    }
+    return network_with_ids;
+  }
+
+ private:
+  // Translate from a callnode to the appropriate 'Make' method
+  sl::TensorsAndId HandleCall(const CallNode*);
+
+  void VisitExpr_(const CallNode* cn) final;
+  void VisitExpr_(const TupleNode* op) final;
+  void VisitExpr_(const TupleGetItemNode* tg) final;
+  void VisitLeaf(const Expr& expr) final;
+
+  // Make a support library operand from a Call
+  EthosnError MakeConcatenateLayer(const Call& call, sl::TensorAndId<sl::Operand>* out);
+  EthosnError MakeSplitLayer(const Call& call, sl::TensorsAndId* outs);
+
+  /*! \brief A look-up table from Expr to layers. */
+  std::map<Expr, std::vector<std::shared_ptr<sl::Operand>>> operand_table_;
+  /*! \brief A look-up table from Expr to SL operation IDs. */
+  std::map<Expr, std::vector<std::pair<uint32_t, uint32_t>>> id_table_;
+  /*! \brief A look-up table from Expr to tensor infos. */
+  std::map<Expr, std::vector<sl::TensorInfo>> tensor_table_;
+  /*! \brief The support library network to compile. */
+  std::shared_ptr<sl::Network> network_;
+};
+
+NetworkWithIDs ConstructNetwork(const IRModule& mod, const GlobalVar& var, const Function& func) {
+  return ConstructNetworkVisitor(mod, var).Construct(func);
+}
+
+class EthosnCompiler {
+ public:
+  static runtime::ethosn::OrderedCompiledNetwork CompileEthosnFunc(const IRModule& mod,
+                                                                   std::string name,
+                                                                   const Function& func) {
+    // Construct the network
+    GlobalVar var = mod->GetGlobalVar(name);
+    auto network_with_ids = ConstructNetwork(mod, var, func);
+    // Now set the required build flags
+    sl::CompilationOptions options = EthosnAPI::CreateOptions();
+    // Finally compile the network
+    auto compiled_network = EthosnAPI::Compile(network_with_ids.network, options);
+    auto input_output_order = GetInputOutputOrder(network_with_ids, compiled_network);
+    runtime::ethosn::OrderedCompiledNetwork ordered_network;
+    ordered_network.name = name;
+    ordered_network.cmm = std::move(compiled_network);
+    ordered_network.inputs = input_output_order.first;
+    ordered_network.outputs = input_output_order.second;
+    return ordered_network;
+  }
+
+  static runtime::Module CreateRuntimeModule(const ObjectRef& ref) {
+    std::vector<runtime::ethosn::OrderedCompiledNetwork> cmms;
+    if (ref->IsInstance<FunctionNode>()) {
+      IRModule mod;
+      Function bfunc = Downcast<Function>(ref);
+      auto name_node = bfunc->GetAttr<String>(tvm::attr::kGlobalSymbol);
+      CHECK(name_node.defined()) << "Failed to retrieved external symbol.";
+      mod->Add(GlobalVar(name_node.value()), bfunc);
+      for (const auto& it : mod->functions) {
```

Comment:
I can guess why it is written this way, but now that we always get `FunctionNode` here, creating a single function module and iterating over it is unnecessary.